### DISCUSSION TITLE
Why should I switch from fasd?

### DISCUSSION THREAD

**Discussion by joshmedeski** [ajeetdsouza/zoxide#135] [Q&A] [answered]
z, autojump, fasd, and others offer similar features as zoxide. What makes this project different? I see it's mostly written in Rust (compared to fasd being in shell).

Why would I make the switch? Is it faster or does it have better features? The fact that it's in active development and has fzf support built-in is cool.

I think a discussion about why it stands out from the competition would help people who are already using a similar solution (like myself) should switch to zoxide.

p.s. I'm happy to update the README with a summary of this discussion :)

**Comment by ajeetdsouza** [comment, marked as answer]
Hey @joshmedeski, sorry for the delayed response.

- In terms of the algorithm used, autojump weights a directory according to the time spent in that directory. Personally, I prefer zoxide's algorithm - it cd's into a directory based on how many times (and how recently) you have cd'd into it in the past. If I cd into `dir1` once a day and leave the terminal open for the rest of the day, but cd into `dir2` frequently and close the terminal, I think `z dir` should take me to `dir2`, since it would be the correct answer more often.
- In terms of speed, zoxide tends to be orders of magnitude faster than other autojumpers - I like shell plugins to use as little time as possible to avoid slow startup and prompts. If you like, you can use `hyperfine` to test this for yourself.
- zoxide only deals with directories, fasd can open files for you as well. This again boils down to personal preference - I think it's better to cd into a folder and open a file unambiguously from there, but fasd lets you directly query a file path and open it up with the desired application (`mplayer`, `vim`, etc). This works well for some users, but I'd rather not have to exit Vim every time my query returns the wrong answer.
- zoxide's own CLI is designed to be user-friendly and scriptable. A lot of people just use zoxide for the querying engine and write their own logic for z, etc.

If you're on the fence, I'd suggest giving zoxide a shot anyway and seeing if you like it. Generally we're quite open to feature requests here, so if there's something you feel is missing, feel free to create an issue!

**Comment by joshmedeski** [comment]
Thanks @ajeetdsouza, I got a new computer and decided to give zoxide a try, it’s great!

I mostly just used `fasd` to jump between directories, so this is a good substitute.

p.s. built-in fish support works flawlessly, thanks for supporting it!

**Reply by ajeetdsouza** [reply]
Thank you, I'm glad you liked it!

**Comment by PrayagS** [comment]
> zoxide only deals with directories, fasd can open files for you as well. This again boils down to personal preference - I think it's better to cd into a folder and open a file unambiguously from there, but fasd lets you directly query a file path and open it up with the desired application (mplayer, vim, etc). This works well for some users, but I'd rather not have to exit Vim every time my query returns the wrong answer.

@ajeetdsouza I completely agree with your opinion on this but are there any plans on supporting interactive file selection (something like `zif`)? That would take away the problem of matching the wrong file. And after selection, the file is opened using `xdg-open` which makes it such that we neither have to bake in the logic to open a file based on its type and nor implement any more flags or aliases.

Please let me know your thoughts on this. I might even take this up as an opportunity to learn Rust and contribute to implementing this feature.

**Reply by ajeetdsouza** [reply]
@PrayagS I appreciate the suggestion, but I'd still say tracking files is out of scope right now. Sorry!
